queries/orderbook/trade_data.sql

```diff
@@ -0,0 +1,88 @@
+WITH trade_data_raw AS MATERIALIZED (
+    SELECT
+        -- settlement data
+        concat('0x', encode(ss.winner, 'hex')) AS winning_solver,
+        s.auction_id,
+        CASE WHEN s.tx_hash IS NULL THEN NULL
+            ELSE concat('0x', encode(s.tx_hash, 'hex'))
+        END AS tx_hash,
+        -- order data
+        concat('0x', encode(o.uid, 'hex')) AS order_uid,
+        o.kind,
+        o.partially_fillable,
+        concat('0x', encode(o.sell_token, 'hex')) AS sell_token,
+        concat('0x', encode(o.buy_token, 'hex')) AS buy_token,
+        o.sell_amount as limit_sell_amount,
+        o.buy_amount as limit_buy_amount,
+        concat('0x', encode(ad.full_app_data, 'hex')) AS app_data,
+        -- quote data
+        oq.sell_amount as quote_sell_amount,
+        oq.buy_amount as quote_buy_amount,
+        oq.gas_amount as quote_gas_amount,
+        oq.gas_price as quote_gas_price,
+        oq.sell_token_price as quote_sell_token_price,
+        CASE WHEN oq.solver IS NULL THEN NULL
+            ELSE concat('0x', encode(oq.solver, 'hex'))
+        END AS quote_solver,
+        -- trade data
+        t.sell_amount, -- the total amount the user sends
+        t.buy_amount, -- the total amount the user receives
+        oe.surplus_fee as observed_fee, -- the total discrepancy between what the user sends and what they would have send if they traded at clearing price
+        -- native prices
+        ap_sell.price as sell_token_native_price,
+        ap_buy.price as buy_token_native_price
+    FROM
+        settlements s
+        JOIN settlement_scores ss -- contains block_deadline
+        ON s.auction_id = ss.auction_id
+        JOIN trades t -- contains traded amounts
+        ON s.block_number = t.block_number -- log_index cannot be checked, does not work correctly with multiple auctions on the same block
+        JOIN orders o -- contains tokens and limit amounts
+        ON t.order_uid = o.uid
+        JOIN order_execution oe -- contains surplus fee
+        ON t.order_uid = oe.order_uid
+        AND s.auction_id = oe.auction_id
+        LEFT OUTER JOIN order_quotes oq -- contains quote amounts
+        ON o.uid = oq.order_uid
+        LEFT OUTER JOIN app_data ad
+        ON o.app_data = ad.contract_app_data 
+        LEFT OUTER JOIN auction_prices ap_sell -- contains price: sell token
+        ON s.auction_id = ap_sell.auction_id
+        AND o.sell_token = ap_sell.token
+        LEFT OUTER JOIN auction_prices ap_buy -- contains price: sell token
```

Comment:
Have to admit i  prefer to break down such queries into multiple intermediate tables as this is not trivial to parse for me.

Reply:
I personally only like to break up the query if some computation becomes easier. Here we are just joining the tables one by one without any additional computation.

Do you have a certain split in mind?

Reply:
>>Do you have a certain split in mind?

Not really, tbh. I read again and it is actually not hard to follow. Just looks intimidating i guess at a first glance, and hard to be fully convinced it works as expected.